### PR TITLE
Separating CaptureDataFrame out of DFIterDataPipe

### PR DESCRIPTION
**Pull request by VitalyFedyunin**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71776

Differential Revision: [D33771602](https://our.internmc.facebook.com/intern/diff/D33771602)

### CODE REVIEW

**Review by ejguan — APPROVED** [review]

**Review by wenleix — APPROVED** [review]
Thanks!

Can we visualize the `CaputreDataFrame`? -- e.g. some kind of a expression tree or IR nodes, etc.

**Comment by VitalyFedyunin** [comment]
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33771602).

**Comment by VitalyFedyunin** [comment]
> Thanks!
> 
> Can we visualize the `CaputreDataFrame`? -- e.g. some kind of a expression tree or IR nodes, etc.

`str(CaptureDataFrame)`

File: torch/utils/data/datapipes/dataframe/dataframes.py [hunk 1]
```diff
@@ -25,16 +27,16 @@ def __iter__(self):
...
 
     def __init__(self):
         self.ctx = {'operations': [], 'variables': []}
 
     def __str__(self):
-        return self.ops_str()
+        return self._ops_str()
 
-    def ops_str(self):
+    def _ops_str(self):
```
Comment (VitalyFedyunin) [inline torch/utils/data/datapipes/dataframe/dataframes.py:39]: @wenleix this function returns graph in readable form

**Pull request closed without merging**